libraries/chain/include/steem/chain/database.hpp

```diff
@@ -33,6 +34,15 @@ namespace steem { namespace chain {
    using steem::protocol::price;
    using abstract_plugin = appbase::abstract_plugin;
 
+   class database;
+   using set_index_type_func = std::function< void(database&, mira::index_type, const boost::filesystem::path&, const boost::any&) >;
+
+   struct index_delegate {
+      set_index_type_func set_index_type;
```

Comment:
Why not simply pass around `std::function`s instead of a struct containing the `std::function`? This would allow (if wanted) for the index delegates to be lambdas.

Reply:
The purpose of this was extensibility, if you needed any functionality by index name you could add a function to the struct and modify the macro and every index would have the new feature.